Dfe.Data.SearchPrototype/Web/Models/ViewModels/Shared/Pagination.cs

```
@@ -0,0 +1,118 @@
+
+        /// <summary>
+        /// Determines whether the current page falls within the lower paging
+        /// boundary, given the total number of pages.
+        /// </summary>
+        public bool PageSequenceIncludesFirstPage =>
+            _pager.PageSequenceIncludesFirstPage(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Determines whether the current page falls within the lower paging threshold.
+        /// </summary>
+        public bool HasMoreLowerPagesAvailable =>
+            _pager.HasMoreLowerPagesAvailable(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Determines whether the current page falls within the upper paging
+        /// boundary, given the total number of pages provisioned.
+        /// </summary>
+        public bool PageSequenceIncludesLastPage =>
+            _pager.PageSequenceIncludesLastPage(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Determines whether the current page falls within the upper paging threshold.
+        /// </summary>
+        public bool HasMoreUpperPagesAvailable =>
+            _pager.HasMoreUpperPagesAvailable(CurrentPageNumber, TotalNumberOfPages);
+
+        /// <summary>
+        /// Gets the total number of pages.
+        /// </summary>
+        /// <returns>
+        /// The calculated number of pages available.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// Thrown if either we have a total record count of zero, or records per-page
+        /// configured to zero.
+        /// </exception>
+        private int GetTotalNumberOfPages()
+        {
+            if (TotalRecordCount == 0) {
+                throw new ArgumentException("The record count must be greater than zero.");
+            }
+
+            if (RecordsPerPage == 0){
+                throw new ArgumentException("The page size must be greater than zero.");
+            }
```

if we need to throw an exception about this, it's probably too late to do it here